### ISSUE TITLE
Route remote done signal in PMU control

### ISSUE THREAD

**Issue by yaqiz01** [yaqiz01/pir#25]
PMU's readDone and writeDone could be remotely routed to PMU at the moment, which can cause read stage/write stage be enabled to #ctr iteration + round trip delay between copier and source of the Counter Chain. This is because readEn is locally derived while readDone is remotely routed (example. `SimpleSequentialTest`). Although writeEn is the valid of the vector bus, which is remotely routed, if the vector data bus takes different # hops then routed writeDone, the write address counter can also be enabled for incorrect # of cycles. Therefore, both readDone and writeDone have to be locally derived and read counter chain and write counter chain have to be copied if necessary. 
In summary, remote done routing should only be allowed if enable of the same counter is not used in receiver CU. 

TODO:
Fix control allocation and mapping
Remove input from tokenIn in readDoneXbar and writeDoneXbar
change swapWriteXbar to select from readDone and writeDone

**Comment by raghup17** [comment]
Sounds good.

**Comment by yaqiz01** [comment]
Fixed in spade and pir

**Comment by yaqiz01** [comment]
Fixed in spade and pir

**Comment by yaqiz01** [comment]
Fixed in spade and pir
